clang/lib/Lex/PPExpressions.cpp

```diff
@@ -333,13 +333,14 @@ static bool EvaluateValue(PPValue &Result, Token &PeekTok, DefinedTracker &DT,
                                  : diag::ext_cxx23_size_t_suffix
                            : diag::err_cxx23_size_t_suffix);
 
-    // 'wb/uwb' literals are a C23 feature. We explicitly do not support the
-    // suffix in C++ as an extension because a library-based UDL that resolves
-    // to a library type may be more appropriate there.
+    // 'wb/uwb' literals are a C23 feature.
+    // '__wb/__uwb' are a C++ extension.
     if (Literal.isBitInt)
-      PP.Diag(PeekTok, PP.getLangOpts().C23
-                           ? diag::warn_c23_compat_bitint_suffix
-                           : diag::ext_c23_bitint_suffix);
+      PP.Diag(PeekTok, !PP.getLangOpts().CPlusPlus
+                           ? PP.getLangOpts().C23
+                                 ? diag::warn_c23_compat_bitint_suffix
+                                 : diag::ext_c23_bitint_suffix
+                           : diag::ext_cpp_bitint_suffix);
```

Comment:
```suggestion
      PP.Diag(PeekTok, PP.getLangOpts().CPlusPlus ? diag::ext_cpp_bitint_suffix
                     : PP.getLangOpts().C23       ? diag::warn_c23_compat_bitint_suffix
                     :                              diag::ext_c23_bitint_suffix);
```
The nested ternaries are a bit much, in my opinion; I’d rewrite it to something like this instead (also make sure to run clang-format on this afterwards, I’ve just manually aligned them here so it’s clear what change I’m proposing here).